### PR TITLE
Use Snapping endpoint

### PR DESCRIPTION
**Pull request by mrkhan**
Basic snapping functionality with
- added test case to request for json only
- added test case with driving profile

### CODE REVIEW

File: src/OrsSnap.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        }
+    }
+
+    getBody(args) {
+        if (args.options && typeof args.options !== 'object') {
+            args.options = JSON.parse(args.options)
+        }
+
+        return args
+    }
```
Comment (TheGreatRefrigerator) [inline src/OrsSnap.js:21]: I think we dont need this. snap doesn't have an options parameter.

File: src/__tests__/OrsSnap.cy.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+    context('construction', () => {
+        it('sets correct service and API version', () => {
+            expect(orsSnap.defaultArgs.service).to.equal('snap')
+            expect(orsSnap.defaultArgs.api_version).to.equal('v2')
+        })
+    })
+
+    context('methods', () => {
+        context('getBody', () => {
```
Comment (TheGreatRefrigerator) [inline src/__tests__/OrsSnap.cy.js:16]: we don't need this either

File: src/__tests__/OrsSnap.cy.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+            it('snap response has snapped_distance', (done) => {
+                orsSnap.calculate({
+                    locations: [[8.681495,49.51461],[8.686507,49.41943]],
+                    radius: 300,
+                    profile: 'driving-car',
+                    format: 'json'
+                }).then((json) => {
+                    expect(json['locations'].length).to.be.greaterThan(0)
+                    expect(json['locations'][1].snapped_distance).exist
```
Comment (TheGreatRefrigerator) [inline src/__tests__/OrsSnap.cy.js:43]: existance of snapped distance can be tested together with the simple snap result.
I would rather have another test for not found entries (null,[location])

File: src/__tests__/OrsSnap.cy.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+            it('snap response has snapped_distance in geojson format', (done) => {
+                orsSnap.calculate({
+                    locations: [[8.681495,49.51461],[8.686507,49.41943]],
+                    radius: 300,
+                    profile: 'driving-car',
+                    format: 'geojson'
+                }).then((json) => {
+                    expect(json['features'].length).to.be.greaterThan(0)
+                    expect(json['features'][0].properties.snapped_distance).exist
```
Comment (TheGreatRefrigerator) [inline src/__tests__/OrsSnap.cy.js:70]: see above

**Review by TheGreatRefrigerator — CHANGES_REQUESTED** [review]
I would put the dist file build in a separate commit as well as the tests.
